hackathon_site/event/api_views.py

```
@@ -66,3 +81,57 @@ def post(self, request, *args, **kwargs):
         response_serializer = TeamSerializer(profile.team)
         response_data = response_serializer.data
         return Response(data=response_data, status=status.HTTP_201_CREATED,)
+
+
+class JoinTeamView(generics.GenericAPIView, mixins.RetrieveModelMixin):
+    permission_classes = [UserHasProfile]
+    serializer_class = TeamSerializer
+
+    def clean(self):
+        if not is_registration_open():
+            raise forms.ValidationError(
+                _("You cannot change teams after registration has closed."),
+                code="registration_closed",
+            )
+
+        return super().clean()
+
+    @transaction.atomic
+    def post(self, request, team_code, *args, **kwargs):
+
+        profile = request.user.profile
+        current_team = profile.team
+
+        try:
+            team = Team.objects.get(team_code=team_code)
+        except:
```

It's bad practice to have a generic `except`, since it may swallow errors that we haven't thought about (your IDE should be warning you of this).

Instead, do `except Team.DoesNotExist`, which is the error that will be thrown if the team doesn't exist.

---

resolved in new code